radio/src/sdcard.cpp

```diff
@@ -40,21 +40,70 @@ const char * sdCheckAndCreateDirectory(const char * path)
   return NULL;
 }
 
-bool isFileAvailable(const char * path)
+bool isFileAvailable(const char * path, bool exclDir)
 {
+  if (exclDir) {
+    FILINFO fno;
+    return (f_stat(path, &fno) == FR_OK && !(fno.fattrib & AM_DIR));
+  }
   return f_stat(path, 0) == FR_OK;
 }
 
-bool isFileAvailable(const char * filename, const char * directory)
+/**
+  Search file system path for a file. Can optionally take a list of file extensions to search through.
+  Eg. find "splash.bmp", or the first occurrence of one of "splash.[bmp|jpeg|jpg|gif]".
+
+  @param path String with path name, no trailing slash. eg; "/BITMAPS"
+  @param file String containing file name to search for, with or w/out an extension.
+    eg; "splash.bmp" or "splash"
+  @param pattern Optional list of one or more file extensions concatenated together, including the period(s).
+    The list is searched backwards and the first match, if any, is returned.  If null, then only the actual filename
+    passed will be searched for.
+    eg: ".gif.jpg.jpeg.bmp"
+  @param exclDir true/false whether to allow directory matches (default true, excludes directories)
+  @param match Optional container to hold the matched file extension (wide enough to hold LEN_FILE_EXTENSION_MAX + 1).
+  @retval true if a file was found, false otherwise.
+*/
+static bool isFilePatternAvailable(const char * path, const char * file, const char * pattern = NULL, bool exclDir = true, char * match = NULL)
```

Comment:
Really just marking it as "local" function to make it obvious it is not (currently) being used outside the file...  certainly doesn't have to be there, more of a style thing I guess.